comp/core/agenttelemetry/impl/agenttelemetry.go

```diff
@@ -128,9 +129,13 @@ func createAtel(
 
 // NewComponent creates a new agent telemetry component.
 func NewComponent(req Requires) agenttelemetry.Component {
+	if !pkgconfigsetup.IsAgentTelemetryEnabled(req.Config) {
+		return &atel{enabled: false}
+	}
+
 	sender, err := createSender(req.Config, req.Log)
 	if err != nil {
-		return &atel{}
+		return &atel{enabled: false}
```

Comment:
Would it make sense to log the error that prevented telemetry from being enabled?

Reply:
Indirectly addressed, thank you